test/Frontend/Desugar.hs

```diff
@@ -24,78 +16,53 @@ allDesugar =
     "desugar Tests"
     [guardTest]
 
-shouldDesugar ::
-  T.TestName -> ByteString -> [Desugared.TopLevel] -> T.TestTree
+shouldDesugar :: T.TestName -> ByteString -> [Desugared.TopLevel] -> T.TestTree
 shouldDesugar name x y =
   T.testGroup
     "Desugar tests"
     [ T.testCase
         ("desugar: " <> name <> " " <> show x <> " should desugar to " <> show y)
-        (fmap desugar (parseOnly (many Parser.topLevelSN) x) T.@=? Right y)
+        ( fmap Desugar.f (Parsec.parseOnly (many Parser.topLevelSN) x)
```

Comment:
```suggestion
        ( fmap Desugar.desugar (Parsec.parseOnly (many Parser.topLevelSN) x)
```
easier to see that you are desugaring. `f` is not a good name.

Reply:
no, f is the main function of the desugar module, I use it in other places.

Much like how one writes `NameSymbol.T` and not `NameSymbol.NameSymbol`

Reply:
Resolved to use:

`op` - stateless transformations
`exec` - stateful IO somethings